### PR TITLE
Upgrade apollo server packages

### PR DESCRIPTION
**Pull request by timleslie**
Details of the changes between versions 2 and 3 can be found here: https://www.apollographql.com/docs/apollo-server/migration/

### CODE REVIEW

File: packages/keystone/src/lib/server/createApolloServer.ts [hunk 1]
```diff
@@ -61,71 +60,13 @@ const _createApolloServerConfig = ({
...
-  } else if (typeof apolloConfigIntrospection === 'boolean') {
-    // graphql.apolloConfig.introspection === true (introspection accessible in all cases)
-    introspection = true;
-  } else if (process.env.NODE_ENV === 'production') {
-    // process.env.NODE_ENV === 'production' (introspection not accessible in production)
-    introspection = undefined;
-  } else if ((typeof playground === 'boolean' && playground) || typeof playground === 'object') {
-    // not specified at all (introspection enabled if playground is enabled or defined)
-    introspection = true;
-  }
```
Comment (mitchellhamilton) [inline packages/keystone/src/lib/server/createApolloServer.ts:105]: What's the implication of this deleted code?

File: docs/pages/docs/walkthroughs/embedded-mode-with-sqlite-nextjs.mdx [hunk 2]
```diff
@@ -282,15 +282,17 @@ Run `yarn dev` again.
 
 ## Bonus: add the GraphQL API to the frontend
 
-To get a read-only GraphQL API and playground in production, add `/pages/api/graphql.tsx`with the following:
+FIXME
```
Comment (mitchellhamilton) [inline docs/pages/docs/walkthroughs/embedded-mode-with-sqlite-nextjs.mdx:285]: ?

File: docs/pages/why-keystone.tsx [hunk 3]
```diff
@@ -521,7 +521,7 @@ export default function WhyKeystonePage() {
             <Type as="p" look="body18" color="var(--muted)" margin="1rem 0 0 0">
               Use an API-first content platform to unlocks the power of multichannel content ops.
               Connect to your audience where they want to be. Ship content to any frontend over
-              performant APIs. Use the built-in GraphQL playground to query with ease.
+              performant APIs. Use the built-in Apollo Playground to query with ease.
```
Comment (mitchellhamilton) [inline docs/pages/why-keystone.tsx:524]:
```suggestion
              performant APIs. Use the built-in Apollo Sandbox to query with ease.
```

File: packages/keystone/src/lib/server/createExpressServer.ts [hunk 4]
```diff
@@ -35,7 +35,13 @@ const addApolloServer = ({
   server.use(graphqlUploadExpress({ maxFileSize }));
   // FIXME: Support custom API path via config.graphql.path.
   // Note: Core keystone uses '/admin/api' as the default.
-  apolloServer.applyMiddleware({ app: server, path: '/api/graphql', cors: false });
+  await apolloServer.start();
+  apolloServer.applyMiddleware({
+    app: server,
+    path: '/api/graphql',
+    // FIXME: Allow this to be set through config
+    cors: { origin: 'https://studio.apollographql.com', credentials: true },
```
Comment (mitchellhamilton) [inline packages/keystone/src/lib/server/createExpressServer.ts:43]: Could we not allow this in production?

Not really a thing to action but it's kinda annoying this is on a different origin to the API which means you can't sign in to the Admin UI and then use that logged in state while exploring the GraphQL API, you'd have to do the mutation to sign in 😞

**Review by emmatown — CHANGES_REQUESTED** [review]

**Review by emmatown — APPROVED** [review]

**Pull request merged**